vitrina/api/views.py

```diff
@@ -1,56 +1,479 @@
+DATASET_ID = openapi.Parameter('datasetId', in_=openapi.IN_PATH, type=openapi.TYPE_INTEGER)
+DISTRIBUTION_ID = openapi.Parameter('distributionId', in_=openapi.IN_PATH, type=openapi.TYPE_INTEGER)
+STRUCTURE_ID = openapi.Parameter('structureId', in_=openapi.IN_PATH, type=openapi.TYPE_INTEGER)
 
 
-@method_decorator(name='list', decorator=swagger_auto_schema(
-    operation_summary="Retrieve a list of available catalogs",
-    operation_description="List of catalogs",
-    manual_parameters=[header_param],
-    tags=[CATALOG_TAG]
-))
 class CatalogViewSet(ListModelMixin, GenericViewSet):
     serializer_class = CatalogSerializer
     queryset = Catalog.objects.all()
     permission_classes = (APIKeyPermission,)
 
+    @swagger_auto_schema(
+        operation_summary="Retrieve a list of available catalogs",
+        operation_description="List of catalogs",
+        manual_parameters=[HEADER_PARAM],
+        tags=[CATALOG_TAG]
+    )
+    def list(self, request, *args, **kwargs):
+        return super().list(request, *args, **kwargs)
+
 
-@method_decorator(name='list', decorator=swagger_auto_schema(
-    operation_summary="Retrieve all available categories",
-    operation_description="All the categories",
-    manual_parameters=[header_param],
-    tags=[CATEGORY_TAG]
-))
 class CategoryViewSet(ListModelMixin, GenericViewSet):
     serializer_class = CategorySerializer
     queryset = Category.objects.all()
     permission_classes = (APIKeyPermission,)
 
+    @swagger_auto_schema(
+        operation_summary="Retrieve all available categories",
+        operation_description="All the categories",
+        manual_parameters=[HEADER_PARAM],
+        tags=[CATEGORY_TAG]
+    )
+    def list(self, request, *args, **kwargs):
+        return super().list(request, *args, **kwargs)
+
 
-@method_decorator(name='list', decorator=swagger_auto_schema(
-    operation_summary="Retrieve a list of available licences",
-    operation_description="All the licences",
-    manual_parameters=[header_param],
-    tags=[LICENCE_TAG]
-))
 class LicenceViewSet(ListModelMixin, GenericViewSet):
     serializer_class = LicenceSerializer
     queryset = Licence.objects.all()
     permission_classes = (APIKeyPermission,)
+
+    @swagger_auto_schema(
+        operation_summary="Retrieve a list of available licences",
+        operation_description="All the licences",
+        manual_parameters=[HEADER_PARAM],
+        tags=[LICENCE_TAG]
+    )
+    def list(self, request, *args, **kwargs):
+        return super().list(request, *args, **kwargs)
+
+
+class DatasetViewSet(RevisionMixin, ModelViewSet):
+    serializer_class = DatasetSerializer
+    permission_classes = (APIKeyPermission,)
+    lookup_url_kwarg = 'datasetId'
+
+    def get_queryset(self):
+        organization = get_api_key_organization(self.request)
+        if organization:
+            return Dataset.objects.filter(organization=organization)
```

Comment:
```suggestion
            return Dataset.objects.filter(organization=organization, deleted__isnull=True)
```

We need to exclude deleted datasets.